### PR TITLE
CASMCMS-8069: Barebones boot test now fails if user requested compute node is not available

### PR DESCRIPTION
**Pull request by mharding-hpe**
# Description

This PR updates the test information to reflect this fact. It also performs a bunch of linting and reorganization of the information on this test.

This should not merge until the manifest PRs for the test change merge.

## Issues and Related PRs

https://github.com/Cray-HPE/csm/pull/1059
https://github.com/Cray-HPE/csm-rpms/pull/513

# Checklist Before Merging

- [X] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [X] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [X] My commits or Pull-Request Title contain my JIRA information, or I don't have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Review by rustydb — APPROVED** [review]

File: troubleshooting/cms_barebones_image_boot.md [hunk 1]
```diff
@@ -1,57 +1,97 @@
...
+  * This is true both for the test script and for the [Manual test procedure](#manual-test-procedure).
+* The test script uses the Kubernetes API gateway to access CSM services. The gateway must be properly configured to allow an access token to be generated by the script.
+* The manual procedure uses the Cray CLI. The CLI must be configured and authenticated on the node where the manual steps are being performed.
+  * See [Configure the Cray CLI](../operations/configure_cray_cli.md).
+* The test script is installed as part of the `cray-cmstools-crayctldeploy` RPM.
+
+## Test script
+
+The script file location is `/opt/cray/tests/integration/csm/barebonesImageTest`.
+Be sure to review the [Test prerequisites](#test-prerequisites).
```
Comment (nrockershousen) [inline troubleshooting/cms_barebones_image_boot.md:52]:
```suggestion
Review the [Test prerequisites](#test-prerequisites) before proceeding.
```

File: troubleshooting/cms_barebones_image_boot.md [hunk 2]
```diff
@@ -65,35 +105,36 @@ environment variables. To control the information being sent to the console, set
...
 cray.barebones-boot-test: DEBUG    Found boot image: cray-shasta-csm-sles15sp2-barebones.x86_64-shasta-1.5
 cray.barebones-boot-test: DEBUG    Creating bos session template with etag:bc390772fbe67107cd58b3c7c08ed92d, path:s3://boot-images/e360fae1-7926-4dee-85bb-f2b4eb216d9c/manifest.json
 ```
 
-## 4. Manual Steps To Reproduce This Script
+## Manual test procedure
 
 The following manual steps may be performed to reproduce the actions of this script.
-The result should be the same as running the script.
+Be sure to review the [Test prerequisites](#test-prerequisites) before beginning.
```
Comment (nrockershousen) [inline troubleshooting/cms_barebones_image_boot.md:127]:
```suggestion
Review the [Test prerequisites](#test-prerequisites) before beginning.
```

File: troubleshooting/cms_barebones_image_boot.md [hunk 3]
```diff
@@ -1,57 +1,97 @@
...
 ```bash
 /opt/cray/tests/integration/csm/barebonesImageTest --xname x3000c0s10b1n0
 ```
 
-## 3. Controlling Test Script Output Level
+> Troubleshooting: If any compute nodes are missing from HSM database, then refer to
+> [2.2.2 Known issues with HSM discovery validation](../operations/validate_csm_health.md#222-known-issues-with-hsm-discovery-validation)
+> in order to troubleshoot any node BMCs that have not been discovered.
+
+### Controlling test script output level
```
Comment (nrockershousen) [inline troubleshooting/cms_barebones_image_boot.md:94]:
```suggestion
### Control test script output level
```

**Comment by mharding-hpe** [comment]
When this merges, its other two manifest PRs should merge as well:
https://github.com/Cray-HPE/csm-rpms/pull/513
https://github.com/Cray-HPE/csm/pull/1059

**Review by dlaine-hpe — APPROVED** [review]

**Review by nrockershousen — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Comment by rustydb** [comment]
@mharding-hpe there's a problem with the csm-rpms PR, should this docs-csm PR be reverted?

**Comment by mharding-hpe** [comment]
> @mharding-hpe there's a problem with the csm-rpms PR, should this docs-csm PR be reverted?
No, I can make a stable artifact and update those PRs.

**Pull request merged**